nemo/collections/llm/quantization/quantizer.py

```diff
@@ -0,0 +1,332 @@
+        # Export sanity checks
+        if export_config is not None:
+            assert dtype in SUPPORTED_DTYPE, f"Unsupported export dtype: {dtype}"
+        self.torch_dtype = torch_dtype_from_precision(dtype)
+
+    def load_quantizable_model(self, nemo_checkpoint_path: str, calib_tp: int = 1, calib_pp: int = 1) -> llm.GPTModel:
+        trainer = nl.Trainer(
+            devices=calib_tp * calib_pp,
+            strategy=nl.MegatronStrategy(
+                tensor_model_parallel_size=calib_tp,
+                pipeline_model_parallel_size=calib_pp,
+            ),
+            plugins=nl.MegatronMixedPrecision(precision='16-mixed'),
+        )
+        fabric = trainer.to_fabric()
+        trainer.strategy.setup_environment()
+
+        model = nl.io.load_context(nemo_checkpoint_path).model
+        model.config = self.quantizable_model_config(model.config)
+        model = fabric.load_model(nemo_checkpoint_path, model=model)
+
+        self.nemo_checkpoint_path = nemo_checkpoint_path
+        return model
+
+    def _setup(self, model: llm.GPTModel) -> None:
+        """Setup model for quantization."""
+        # TODO: disable activation checkpointing
+        model.config.vocab_size = model.tokenizer.vocab_size
+        model.config.pipeline_dtype = self.torch_dtype  # TODO: for some reason model.pipeline.dtype does not work
+        model.freeze()
+
+    @staticmethod
+    def quantizable_model_config(model_cfg: llm.GPTConfig) -> llm.GPTConfig:
+        """Modify model config for quantization."""
+
+        model_cfg.transformer_layer_spec = get_gpt_layer_modelopt_spec()
+        if model_cfg.sequence_parallel:
+            logging.warning("Disabling sequence parallelism for quantization...")
+            model_cfg.sequence_parallel = False
+        # Only custom ModelOpt spec is supported for Quantization: this custom spec is largely based on local Megatron-LM
+        # layer definitions to avoid Transformer Engine implementations that are currently not supported.
+        # This layer spec also requires RoPE fusion to be disabled for tensor view operations in attention
+        # layer implementation from megatron/core/transformer/dot_product_attention.py to be functional.
+        model_cfg.name = "modelopt"
+        model_cfg.apply_rope_fusion = False
+        return model_cfg
+
+    def _get_decoder_type(self, config: llm.GPTConfig):
+        return self.export_config.get("decoder_type", None) or get_modelopt_decoder_type(config)
+
+    @staticmethod
+    def _get_unwrapped_mcore_model(model: llm.GPTModel):
+        from megatron.core.models.gpt import GPTModel as MCoreGPTModel
+
+        unwrapped_model = model
+        while not isinstance(unwrapped_model, MCoreGPTModel):
+            unwrapped_model = unwrapped_model.module
+
+        return unwrapped_model
+
+    def quantize(self, wrapped_model: llm.GPTConfig, forward_loop):
```

Comment:
Argument `wrapped_model` is of type "config"?

Could we perhaps name it simply `model`?